lib/libvarnishapi/vsl_cursor.c

```diff
@@ -156,6 +147,19 @@ vslc_vsm_next(const struct VSL_cursor *cursor)
 			return (vsl_end);
 		}
 
+		/* New data observed. Ensure load ordering with the log
+		 * writer. */
+		VRMB();
+
+		if (t == VSL_WRAPMARKER) {
+			/* Wrap around not possible at front */
```

Comment:
I would think the `VRMB()` had to go here (after the `if (t == VSL_WRAPMARKER`):
In `vsl_wrap()`, we prepare the new head and then set the wrap marker. So on the reader side, we should have a read barrier after reading the tag, but before reading the new head.

Reply:
Ah never mind, my comment was without substance. `t` is being read further up in line 138